shell/platform/android/io/flutter/embedding/android/FlutterActivity.java

```diff
@@ -353,6 +357,25 @@ protected boolean shouldAttachEngineToActivity() {
     return true;
   }
 
+  /**
+   * Hook for subclasses to easily provide a custom {@code FlutterEngine}.
+   */
+  @Nullable
+  @Override
```

Comment:
Not clear what's the relationship with these and getFlutterEngine.

Reply:
`getFlutterEngine` has been renamed to `provideFlutterEngine` and its purpose remains unchanged.

Reply:
Talked offline, resolving old comments.